dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -53,7 +54,7 @@ public static Load newLoad(Register register, IExpr address, String mo, int cLin
     }
 
     public static Load newLoad(Register register, IExpr address, String mo) {
-        return newLoad(register, address, mo, -1);
+        return newLoad(register, address, mo, ARCH_PRECISION);
```

Comment:
The `-1` here is a Cline argument, not a precision argument.
This is true for all the methods you changed in `EventFactory`.